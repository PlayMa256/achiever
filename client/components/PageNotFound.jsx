import React from 'react';
import strings from '../../shared/strings';

const PageNotFound = () => (
	<div>{strings.pageNotFound}</div>
);

export default PageNotFound;
