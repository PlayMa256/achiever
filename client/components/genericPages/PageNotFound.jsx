import React from 'react';
import strings from '../../../shared/strings';

const PageNotFound = () => (
	<div className="page-wrapper">
		<h2 className="current-date">
			{' '}
		</h2>
		<div className="column">
			&nbsp;
		</div>
		<div className="column page-not-found">
			{strings.pageNotFound}
			<button
				className="send"
				type="button"
				onClick={() => window.history.back()}
			>
				{strings.goBack}
			</button>
		</div>
	</div>
);

export default PageNotFound;
