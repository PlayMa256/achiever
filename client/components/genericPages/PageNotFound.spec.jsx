import React from 'react';
import { shallow } from 'enzyme';

import PageNotFound from './PageNotFound';

describe('PageNotFound', () => {
	const wrapper = shallow(<PageNotFound />);

	it('should render properly', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('should go back on click button', () => {
		window.history.back = jest.fn();
		const button = wrapper.find('button');
		button.simulate('click', {});
		expect(window.history.back).toHaveBeenCalled();
	});
});
