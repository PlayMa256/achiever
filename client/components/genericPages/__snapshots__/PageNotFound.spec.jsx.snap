// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageNotFound />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="current-date"
        >
           
        </h2>,
        <div
          className="column"
        >
           
        </div>,
        <div
          className="column page-not-found"
        >
          Página não encontrada
          <button
            className="send"
            onClick={[Function]}
            type="button"
          >
            Voltar
          </button>
        </div>,
      ],
      "className": "page-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " ",
          "className": "current-date",
        },
        "ref": null,
        "rendered": " ",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " ",
          "className": "column",
        },
        "ref": null,
        "rendered": " ",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Página não encontrada",
            <button
              className="send"
              onClick={[Function]}
              type="button"
            >
              Voltar
            </button>,
          ],
          "className": "column page-not-found",
        },
        "ref": null,
        "rendered": Array [
          "Página não encontrada",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Voltar",
              "className": "send",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "Voltar",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="current-date"
          >
             
          </h2>,
          <div
            className="column"
          >
             
          </div>,
          <div
            className="column page-not-found"
          >
            Página não encontrada
            <button
              className="send"
              onClick={[Function]}
              type="button"
            >
              Voltar
            </button>
          </div>,
        ],
        "className": "page-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " ",
            "className": "current-date",
          },
          "ref": null,
          "rendered": " ",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " ",
            "className": "column",
          },
          "ref": null,
          "rendered": " ",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Página não encontrada",
              <button
                className="send"
                onClick={[Function]}
                type="button"
              >
                Voltar
              </button>,
            ],
            "className": "column page-not-found",
          },
          "ref": null,
          "rendered": Array [
            "Página não encontrada",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Voltar",
                "className": "send",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": "Voltar",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
